stat/wilcoxontest/wilcoxon_hyp.go

```diff
@@ -0,0 +1,182 @@
+package wilcoxon
+
+import (
+	"gonum.org/v1/gonum/stat/distuv"
+	"math"
+	"sort"
+)
+
+func ensureDataConformance(x []float64, y []float64) {
+	if len(x) == 0 {
+		panic("x array length is zero")
+	}
+	if len(y) == 0 {
+		panic("y array length is zero")
+	}
+	if len(x) != len(y) {
+		panic("dimension mismatch in ensure data conformance")
+	}
+}
+
+func CalculateDifferences(x []float64, y []float64, z []float64) {
+	ensureDataConformance(x, y)
+	ensureDataConformance(x, z)
+	for i := 0; i < len(x); i++ {
+		z[i] = x[i] - y[i]
+	}
+}
+
+func CalculateAbsDifference(x []float64, y []float64, z []float64) {
+	ensureDataConformance(x, y)
+	ensureDataConformance(x, z)
+	for i := 0; i < len(x); i++ {
+		z[i] = math.Abs(y[i] - x[i])
+	}
+}
```

Comment:
This is `f64.LinfDist`.

Reply:
Thanks for review @kortschak , I don't think f64.LinfDist is correct here, LinfDist is calculating the maximum abs difference. Do you want me to call that function with the single element length array (len(x)) times ?. Thanks @kortschak

Reply:
You're correct.